fix: editor-view-react type-check TS2774 (selection-handler)

## Error
`packages/editor-view-react type-check: src/selection-handler.ts(24,35): error TS2774: This condition will always return true since this function is always defined. Did you mean to call it instead?`

## Cause
`contentEditableElement && contentEditableElement.querySelector` — `querySelector` is always defined on HTMLElement, so the condition is always true when element exists.

## Fix
Remove redundant `contentEditableElement.querySelector` check; use `if (contentEditableElement)` only.

--- a/.changeset/fix-editor-view-react-type-check.md
+++ b/.changeset/fix-editor-view-react-type-check.md
@@ -1,0 +1,5 @@
+---
+"@barocss/editor-view-react": patch
+---
+
+Fix type-check TS2774 in selection-handler: remove redundant querySelector check (HTMLElement always has it).

--- a/packages/editor-view-react/src/selection-handler.ts
+++ b/packages/editor-view-react/src/selection-handler.ts
@@ -21,7 +21,7 @@ export class ReactSelectionHandler {
   private _isProgrammaticChange = false;
   private _getScopeRoot(): ParentNode {
     const contentEditableElement = this.getContentEditableElement();
-    if (contentEditableElement && contentEditableElement.querySelector) {
+    if (contentEditableElement) {
       return contentEditableElement;
     }
     return document;

